File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int current_sum = 0;
        for (int i = 0; i<k; i++){
            current_sum += nums[i];
        }
        int max_sum = current_sum;

        int start = 0;
        int end = k;

        while (end < nums.length){
            current_sum += nums[end] - nums[start];
            max_sum = Math.max(current_sum, max_sum);
            start++;
            end++;
        }
        return (double) max_sum / k;

        
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:

        sum_k = 0
        for i in range(k):
            sum_k += nums[i]
        
        max_sum = sum_k

        start = 0
        end = k

        while end < len(nums):
            sum_k += nums[end] - nums[start]
            max_sum = max(max_sum, sum_k)
            start += 1
            end += 1
        
        return max_sum /k



       